### PR TITLE
Kp/ac2 graphs updates

### PR DESCRIPTION
**Pull request by kyra-at-ada**
This PR brings in the changes made to the Graphs topic from AC2, but also makes several updates to address C17 student feedback
**Summary of changes**
- Adds AC2 graphs changes to second two graph lessons within topic
   - Pulls Dijkstra's Graph review questions into lesson between Graphs Overview and introducing BFS/DFS
   - Adds pseudocode for BFS/DFS before code exercises
- Added multiple choice questions addressing input/output for BFS/DFS questions especially for edge cases
    - Addresses feedback from students that the transition from introing graphs to having them write the algorithm was too steep 
- Updated bfs/dfs gifs because I didn't realize how tiny they were in the AC2 lesson!
- Updated the pseudocode for bfs/dfs to use more general language
- Changed the bfs/dfs methods to accept a `start_node` because it better matches our intro to graphs where we talk about wanting to be able to traverse from a given node.
    - Also changed because we give them a dictionary which is technically ordered in Python but we probably don't want them to be in the habit of assuming that for a general hashtable data structure
    - Added paragraph to talk about possibility of choosing a random node in implementation
- More test cases for BFS & DFS to address edge cases
- Added images and explanation for how to access neighbors depending on graph representation
   
**Notes**
- Do we want to split Graph Algorithms up into two lessons? One lesson for BFS and one for DFS. My thinking is now this lesson is quite long so it might be more digestible in two separate lessons, however in the study guide we have them as "Graphs Algorithms" so it might be confusing. However, I do believe in the study guide I have them work on Graphs algorithms through BFS and then through DFS on a different day so it could work out. 
- Updating the pseudocode hopes to address feedback that folks struggled to adapt our basic version of BFS/DFS to other problems

### CODE REVIEW

**Review by char-adadev — APPROVED** [review]
Great work on this! I like the gentle transition from introducing graphs and their representations, to the checks for understanding on different graph representations before going into the traversal algorithms. 

Taking a look through the lesson, I do think it would make sense to break up the Graphs Algorithms lesson into BFS and DFS. I can understand how daunting it may feel to go into a lesson and see a tiny scroll bar - it may even deter some folx from wanting to get started! 

I left a few suggestions for some typos/wording and a small change for two of the coding exercise example implementations. 

Thank you so much!!

File: 04-graphs/03-graphs-algorithms.md [hunk 1]
```diff
@@ -0,0 +1,1227 @@
...
+
+<iframe src="https://adaacademy.hosted.panopto.com/Panopto/Pages/Embed.aspx?id=42ac2142-176c-4846-8b69-af1d012341bf&autoplay=false&offerviewer=true&showtitle=true&showbrand=false&captions=true&interactivity=all" height="360" width="640" style="border: 1px solid #464646;" allowfullscreen allow="autoplay"></iframe>
+
+### !end-hint
+
+### !explanation 
+An example of a working implementation:
+
+```python
+from collections import deque
```
[the remark this answers is not in the document]
Comment (kyra-at-ada) [reply 04-graphs/03-graphs-algorithms.md:329]: Interesting! It does run for me, but I will move it just to make sure!
